packages/dendron-next-server/pages/workspace/config.tsx

```diff
@@ -0,0 +1,326 @@
+import { DendronConfig } from "@dendronhq/common-all";
+import { Formik, Field, Form } from "formik";
+import React, { ReactNode, useState } from "react";
+import {
+  Box,
+  Button,
+  Code,
+  Container,
+  FormControl,
+  FormErrorMessage,
+  FormHelperText,
+  FormLabel,
+  Heading,
+  Input,
+  Stack,
+  Switch,
+  Text,
+  Tooltip,
+} from "@chakra-ui/react";
+import { get } from "lodash";
+
+const getConfigData = (): { data: DendronConfig } => {
+  return {
+    data: {
+      vaults: [
+        {
+          fsPath: "vault",
+          name: "",
+        },
+      ],
+      site: {
+        assetsPrefix: "",
+        copyAssets: false,
+        siteHierarchies: ["dendron"],
+        siteNotesDir: "",
+        siteRepoDir: "",
+        siteRootDir: "docs",
+        usePrettyRefs: false,
+      },
+    },
+  };
+};
+
+// const saveConfigData = (data: DendronConfig) => {
+//   console.log("data saved");
+//   console.log(JSON.stringify(data, null, 2));
+// };
+
+type InputControlProps = {
+  name: string;
+  label?: string;
+  placeholder: string;
+  help: string | ReactNode;
+  disabled?: boolean;
+};
+
+function InputControl({
+  name,
+  label,
+  placeholder,
+  help,
+  disabled,
+}: InputControlProps) {
+  return (
+    <Field name={name}>
+      {({ form: { errors, touched } }) => (
+        <FormControl
+          isInvalid={
+            /**
+             * If field has errors AND they've interacted with this field, mark
+             * control as invalid. */
+            !!(get(errors, name) && get(touched, name))
+          }
+        >
+          {!!label && <FormLabel htmlFor={name}>{label}</FormLabel>}
+
+          <Field
+            //
+            as={Input}
+            name={name}
+            placeholder={placeholder}
+            id={name}
+            disabled={disabled}
+          />
+
+          <FormErrorMessage>{errors.site?.siteRootDir}</FormErrorMessage>
+
+          <FormHelperText>{help}</FormHelperText>
+        </FormControl>
+      )}
+    </Field>
+  );
+}
+
+export default function ConfigSamplePage() {
+  // const { data: configData } = getConfigData();
+
+  const [configData, saveConfigData] = useState(getConfigData().data);
+
+  React.useEffect(() => {
+    console.log("save", JSON.stringify(configData, null, 2));
+  }, [JSON.stringify(configData)]);
+
+  return (
+    <Container paddingBottom={12}>
+      <Stack spacing={4}>
+        <Heading>Config Data</Heading>
+
+        <Formik
+          enableReinitialize
+          initialValues={configData}
+          onSubmit={(formValues) => saveConfigData(formValues)}
```

Comment:
on successful submit, can we show a toast that says "Changes Saved"?

Reply:
:heavy_check_mark: @ 1354b821

![Screenshot from 2020-11-22 12-47-36](https://user-images.githubusercontent.com/595711/99916767-e9a24180-2cc0-11eb-8a82-e53f5dea516d.png)